packages/core/src/lib/pagination/paginate.ts

```diff
@@ -0,0 +1,70 @@
+import { sql } from 'drizzle-orm'
+import type { PgSelect } from 'drizzle-orm/pg-core'
+
+import { QueryParams } from './buildPaginatedUrl'
+import {
+  buildPagination,
+  getPaginationParamsWithDefaults,
+  PaginationArgs,
+} from './buildPagination'
+
+/**
+ * This use $dynamic() query
+ * https://orm.drizzle.team/docs/dynamic-query-building
+ */
+async function paginateQuerySql<T extends PgSelect>({
+  dynamicQuery,
+  page = 1,
+  pageSize = 20,
+}: {
+  dynamicQuery: T
+} & PaginationArgs) {
+  // @ts-ignore
+  dynamicQuery.config.fields = {
+    // @ts-ignore
+    ...dynamicQuery.config.fields,
+    __count: sql<number>`count(*) over()`,
+  }
+  const rows = await dynamicQuery.limit(pageSize).offset((page - 1) * pageSize)
+  const count = rows[0]?.__count ? Number(rows[0]?.__count) : 0
+  return { rows, count }
+}
+
+/**
+ * TODO: Add tests
```

Comment:
👀

Reply:
I know. Do you think it's worth it? In the end this is not so complicated

Reply:
This is a lazy response. I'll do test for it in the next PR